terraform-provider-constellation/internal/provider/attestation_data_source.go

```diff
@@ -0,0 +1,277 @@
+				Computed: true,
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"expected": schema.StringAttribute{
+							Computed: true,
+						},
+						"warn_only": schema.BoolAttribute{
+							Computed: true,
+						},
+					},
+				},
+			},
+			"attestation": schema.SingleNestedAttribute{
+				Computed:            true,
+				MarkdownDescription: "Only relevant for Azure SEV-SNP.",
+				Attributes: map[string]schema.Attribute{
+					"bootloader_version": schema.Int64Attribute{
+						Computed: true,
+					},
+					"tee_version": schema.Int64Attribute{
+						Computed: true,
+					},
+					"snp_version": schema.Int64Attribute{
+						Computed: true,
+					},
+					"microcode_version": schema.Int64Attribute{
+						Computed: true,
+					},
+					"firmware_signer_config": schema.SingleNestedAttribute{
+						Computed: true,
+						Attributes: map[string]schema.Attribute{
+							"accepted_key_digests": schema.ListAttribute{
+								Computed:    true,
+								ElementType: types.StringType,
+							},
+							"enforcement_policy": schema.StringAttribute{
+								Computed: true,
+							},
+							"maa_url": schema.StringAttribute{
+								Computed: true,
+							},
+						},
+					},
+					"amd_root_key": schema.StringAttribute{
+						Computed: true,
+					},
+				},
+			},
+		},
+	}
+}
+
+// Read reads from the data source.
+func (d *AttestationDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data AttestationDataSourceModel
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	csp := cloudprovider.FromString(data.CSP.ValueString())
+	if csp == cloudprovider.Unknown {
+		resp.Diagnostics.AddError("Unknown CSP", fmt.Sprintf("Unknown CSP: %s", data.CSP.ValueString()))
+		return
+	}
+	attestationVariant, err := variant.FromString(data.AttestationVariant.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Unknown Attestation Variant", fmt.Sprintf("Unknown Attestation Variant: %s", data.AttestationVariant.ValueString()))
+		return
+	}
+	if csp == cloudprovider.Azure && attestationVariant.Equal(variant.AzureSEVSNP{}) {
+		snpVersions, err := d.fetcher.FetchAzureSEVSNPVersionLatest(ctx)
+		if err != nil {
+			resp.Diagnostics.AddError("Fetching SNP Version numbers", err.Error())
+			return
+		}
+		tfSnpVersions := convertSNPAttestationTfStateCompatible(resp, snpVersions)
+		diags := resp.State.SetAttribute(ctx, path.Root("attestation"), tfSnpVersions)
+		resp.Diagnostics.Append(diags...)
+		if resp.Diagnostics.HasError() {
+			return
+		}
+	}
+	// TODO align with reenable AWS PR
+	if attestationVariant.Equal(variant.AWSSEVSNP{}) {
+		resp.Diagnostics.AddWarning("AWS attestation not supported", "AWS is not supported yet")
+	}
+
+	verifyFetcher, err := measurements.NewVerifyFetcher(d.client)
```

Comment:
`NewVerifyFetcher` hard codes `sigstore.NewCosignVerifier` as its verifier, meaning we can't test this function.
We should use the same constructor as is used in the `fetchmeasurements` CLI command instead, which allows setting a custom verifier.
The verifier should be set up in `*AttestationDataSource.Configure`, same as the HTTP client already is.

Reply:
these constructors can err however, and configure doesn't return an Error. 
We only have acceptance tests for the data sources at any rate, so we don't need to mock. The VerifyFetcher is testable through the custom constructor.
What do you think?